From f7803fdff700b75c0c51fb91493f622d665bf748 Mon Sep 17 00:00:00 2001
From: Pit Kleyersburg <pitkley@googlemail.com>
Date: Wed, 2 Nov 2016 11:30:03 +0100
Subject: [PATCH] Add config option to specify journal path

---
 beat/config.go      | 1 +
 beat/journalbeat.go | 8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/beat/config.go b/beat/config.go
index de6cd66..9a49046 100644
--- a/beat/config.go
+++ b/beat/config.go
@@ -25,6 +25,7 @@ type JournalReaderConfig struct {
 	CleanFieldNames      *bool   `config:"clean_field_names"`
 	MoveMetadataLocation *string `config:"move_metadata_to_field"`
 	DefaultType          *string `config:"default_type"`
+	JournalPath          *string `config:"journal_path"`
 }
 
 // ConfigSettings holds JournalConfig at the Input section of the config file
diff --git a/beat/journalbeat.go b/beat/journalbeat.go
index 655e7f4..dffa474 100644
--- a/beat/journalbeat.go
+++ b/beat/journalbeat.go
@@ -50,6 +50,7 @@ type Journalbeat struct {
 	cleanFieldnames      bool
 	moveMetadataLocation string
 	defaultType          string
+	journalPath          string
 
 	jr   *sdjournal.JournalReader
 	done chan int
@@ -130,6 +131,12 @@ func (jb *Journalbeat) Config(b *beat.Beat) error {
 		jb.defaultType = "journal"
 	}
 
+	if jb.JbConfig.Input.JournalPath != nil {
+		jb.journalPath = *jb.JbConfig.Input.JournalPath
+	} else {
+		jb.journalPath = ""
+	}
+
 	if _, ok := SeekPositions[jb.seekPosition]; !ok {
 		errMsg := "seek_position must be either cursor, head, or tail"
 		logp.Err(errMsg)
@@ -199,6 +206,7 @@ func (jb *Journalbeat) Setup(b *beat.Beat) error {
 
 	jr, err := sdjournal.NewJournalReader(sdjournal.JournalReaderConfig{
 		Since: time.Duration(1),
+		Path: jb.journalPath,
 		//          NumFromTail: 0,
 	})
 	if err != nil {
-- 
2.7.4

